google/cloud/security/scanner/scanners/base_scanner.py

```diff
@@ -42,14 +45,47 @@ def __init__(self, global_configs, snapshot_timestamp):
             None
         """
         self.global_configs = global_configs
+        self.scanner_configs = scanner_configs
         self.snapshot_timestamp = snapshot_timestamp
+        self.rules = rules
 
     @abc.abstractmethod
     def run(self):
         """Runs the pipeline."""
         pass
 
     @abc.abstractmethod
+    def _retrieve(self, **kwarg):
+        """Runs the pipeline."""
+        pass
+
+
+    @abc.abstractmethod
     def find_violations(self, **kwarg):
```

Comment:
Since you are at it, would it make sense to address this TODO [https://github.com/GoogleCloudPlatform/forseti-security/blob/master/google/cloud/security/scanner/audit/cloudsql_rules_engine.py#L57](url)